src/main/java/Interactors/IOController.java

```diff
@@ -68,18 +68,166 @@ public void connectLogic(){
+                break;
+            case "AlreadyRolled":
+                outputString.append("You already rolled this round! ");
+                break;
+            case "CallAction":
+                outputString.append("You rolled a ").append(currentState.getRoll()).append(currentState.getDescription());
+                break;
+            case "EmptyPropertySpace":
+                outputString.append("You rolled a ").append(currentState.getRoll())
+                        .append(" You have landed on ").append(currentState.getCurrentPlayerProperty().getName())
+                        .append(" and no ones owns this. It costs ")
+                        .append(currentState.getCurrentPlayerProperty().getPrice())
+                        .append(" What do you want to do?");
+                break;
+            case "Buy":
+                //NOTHING
+                break;
+            case "Auction":
+            case "AuctionTree":
+                outputString.append(currentState.getPlayer().getName())
+                        .append(", we are bidding on ").append(currentState.getBiddingProperty().getName())
+                        .append(" with the current pot being ").append(currentState.getBiddingPot());
+                break;
+            case "Steal":
+                outputString.append("What player do you want to steal from?");
+                break;
+            case "ChoosePlayer":
+                //Nothing here
+                break;
+            case "EndTurn":
+                outputString.append("You can't end your turn, you have negative money");
+                break;
+            case "SettingsMenu":
+                outputString.append("Welcome to the settings menu!");
+                break;
+            case "ExitGame":
+                outputString.append("Are you sure you want to exit?");
+                break;
+            case "SaveGame":
+                outputString.append("Game saved!");
+                break;
+            case "Bankruptcy":
+                outputString.append("Confirm bankruptcy?");
+                break;
+            case "Confirmation":
+                //Nothing here
+                break;
+            case "Information":
+                //Nothing here
+                break;
+            case "LowOption":
+                break;
+            case "MediumOption":
+                break;
+            case "HighOption":
+                break;
+            case "Fold":
+                outputString.append(currentState.getPlayer().getName()).append(" won the auction for ")
+                        .append(currentState.getBiddingPot()).append(" dollars");
+                break;
+            case "TradeTree":
+                break;
+            case "AcceptTrade":
+                outputString.append("Trade success");
+                break;
+            case "DeclineTrade":
+                outputString.append("Trade failure");
+                break;
+        }
```

Comment:
I like how you moved all the logic out of the State Entity into IO controller as it emphasizes clean architecture. One thing a little worrying is that IOController might be handling too much now, but this is an issue for another refactor where we create another interface that deals with these values.

Reply:
Exactly, we might want to soon create another class which lies in the outside layers which deal with this handling of state values.